modules/EvseManager/Charger.cpp

```diff
@@ -1014,125 +1020,110 @@ void Charger::setup(bool three_phases, bool has_ventilation, const std::string&
+            (currentState == EvseState::Charging) or (currentState == EvseState::WaitingForEnergy);
+    return (auth and ready);
 }
 
-bool Charger::Authorized_EIM_ready_for_HLC() {
+bool Charger::get_authorized_eim_ready_for_hlc() {
     bool auth = false, ready = false;
-    {
-        std::lock_guard<std::recursive_mutex> lock(configMutex);
-        auth = (authorized && !authorized_pnc);
-    }
-    {
-        std::lock_guard<std::recursive_mutex> lock(stateMutex);
-        ready = (currentState == EvseState::ChargingPausedEV) || (currentState == EvseState::ChargingPausedEVSE) ||
-                (currentState == EvseState::Charging) || (currentState == EvseState::WaitingForEnergy);
-    }
-    return (auth && ready);
+    std::scoped_lock lock(state_machine_mutex);
+    auth = (authorized and !authorized_pnc);
+    ready = (currentState == EvseState::ChargingPausedEV) or (currentState == EvseState::ChargingPausedEVSE) or
+            (currentState == EvseState::Charging) or (currentState == EvseState::WaitingForEnergy);
+    return (auth and ready);
 }
 
-void Charger::Authorize(bool a, const types::authorization::ProvidedIdToken& token) {
+void Charger::authorize(bool a, const types::authorization::ProvidedIdToken& token) {
+    std::scoped_lock lock(state_machine_mutex);
     if (a) {
         // First user interaction was auth? Then start session already here and not at plug in
-        if (!sessionActive())
-            startSession(true);
-        std::lock_guard<std::recursive_mutex> lock(configMutex);
+        if (not session_active)
+            start_session(true);
+
         authorized = true;
         authorized_pnc = token.authorization_type == types::authorization::AuthorizationType::PlugAndCharge;
         id_token = token;
     } else {
-        if (sessionActive()) {
-            stopSession();
+        if (session_active) {
+            stop_session();
         }
-        std::lock_guard<std::recursive_mutex> lock(configMutex);
         authorized = false;
     }
 }
 
-types::authorization::ProvidedIdToken Charger::getIdToken() {
-    std::lock_guard<std::recursive_mutex> lock(configMutex);
+types::authorization::ProvidedIdToken Charger::get_id_token() {
+    std::scoped_lock lock(state_machine_mutex);
     return id_token;
 }
 
-bool Charger::AuthorizedEIM() {
-    std::lock_guard<std::recursive_mutex> lock(configMutex);
-    return (authorized && !authorized_pnc);
-}
-
-bool Charger::AuthorizedPnC() {
-    std::lock_guard<std::recursive_mutex> lock(configMutex);
-    return (authorized && authorized_pnc);
+bool Charger::deauthorize() {
+    std::scoped_lock lock(state_machine_mutex);
+    return deauthorize_internal();
 }
 
-bool Charger::DeAuthorize() {
-    if (sessionActive()) {
-        auto s = getCurrentState();
+bool Charger::deauthorize_internal() {
+    if (session_active) {
+        auto s = currentState;
```

Comment:
currentState is still camelCase while a lot has been changed to snake_case in this PR

Reply:
fixed